src/Entities/Misc/RailedMoveBlock.cs

```diff
@@ -251,9 +251,10 @@ public RailedMoveBlock(Vector2 position, int width, int height, Vector2 node, St
         public override void Update() {
             base.Update();
 
-            bool leftCheck = hasSideButtons && CollideCheck<Player>(Position + new Vector2(-1f, 0f));
-            bool rightCheck = hasSideButtons && CollideCheck<Player>(Position + new Vector2(1f, 0f));
-            bool topCheck = hasTopButtons && CollideCheck<Player>(Position + new Vector2(0f, -1f));
+            bool leftCheck = hasSideButtons && CollideCheck<Player>(Position - Vector2.UnitX);
+            bool rightCheck = hasSideButtons && CollideCheck<Player>(Position + Vector2.UnitX);
+            bool topCheck = hasTopButtons && CollideCheck<Player>(Position - Vector2.UnitY);
+            bool bottomCheck = hasTopButtons && SceneAs<Level>().Session.GetFlag("Xaphan_Helper_Ceiling") && HasPlayerRider();
```

Comment:
Does XaphanHelper patch `HasPlayerRider` so that it also checks above the player?

Reply:
yep!